src/stores/fileDiffs.js

```diff
@@ -23,37 +49,115 @@ export const fileDiffsStore = defineStore('fileDiffs', {
     ...mapActions(appLoadingStore, [
       'setFileDiffsLoading'
     ]),
+    ...mapActions(commitLogStore, [
+      'setUncommitedFilesAmount'
+    ]),
+    clearDiffs: function () {
+      this.setDiffs();
+    },
     setDiffs: function (value) {
       this.diffs = value || [];
-    },
-    getDiffs: async function (currentRepoPath) {
-      if (currentRepoPath) {
-        helpers.setCurrentWorkingDirectory(currentRepoPath);
+      if (this.uncommittedSelected) {
+        this.setUncommitedFilesAmount(this.diffs.length);
       }
-      const command = 'git diff -M --find-copies-harder --histogram';
-      let error;
+    },
+    getUntrackedFileDiff: async function (file) {
+      const command = [
```

Comment:
```suggestion
      /**
       * So this is fun, when using `--no-index` git decided that
       * instead of exit code 0 for OK and 1 for Error, they output
       * 0 if the files match and 1 if they don't. So we need to
       * manually call this command and handle the "error" as a
       * success.
       */
      const command = [
```